src/box/gc.c

```diff
@@ -522,7 +524,18 @@ gc_do_checkpoint(bool is_scheduled)
 	 * Wait the confirms on all "sync" transactions before
 	 * create a snapshot.
 	 */
-	rc = txn_limbo_wait_confirm(&txn_limbo);
+	double timeout;
+	if (replication_synchro_timeout_enabled) {
+		timeout = replication_synchro_timeout;
+	} else {
+		timeout = box_raft()->death_timeout;
+		if (!txn_limbo_is_empty(&txn_limbo)) {
+			say_info("checkpoint will be reached as soon as "
```

Comment:
Let's phrase this a bit differently:
"Checkpoint will be created as soon as all existing synchronous transactions are committed".

In case any of them is rolled back we return an error.

Also I suggest you add this log line regardless of `replication_synchro_timeout_enabled`. It's useful anyway.